pkg/types/spec/validations.go

```diff
@@ -1083,18 +1083,31 @@ func validateONNXPredictor(api *userconfig.API, models *[]CuratedModelResource,
 
 	var modelWrapError func(error) error
 	var modelResources []userconfig.ModelResource
+	var modelFileResources []userconfig.ModelResource
 
 	if hasSingleModel {
-		modelWrapError = func(err error) error {
-			return errors.Wrap(err, userconfig.ModelsPathKey)
+		if strings.HasSuffix(*predictor.Models.Path, ".onnx") {
+			if err := validateONNXModelFilePath(*predictor.Models.Path, projectFiles.ProjectDir(), awsClient, gcpClient); err != nil {
+				return modelWrapError(err)
+			}
+			modelFileResources = []userconfig.ModelResource{
+				{
+					Name: consts.SingleModelName,
+					Path: *predictor.Models.Path,
+				},
+			}
```

Comment:
nit: if lines 1093-1098 are the same as 1100-1105, should they be moved to before the `if` on 1089?

Reply:
Moved it after the if/else block.